src/irc/BridgedClient.ts

```diff
@@ -1144,4 +1135,11 @@ export class BridgedClient extends EventEmitter {
         }
         return this.connectDefer.promise;
     }
+
+    private assertConnected(): Client {
+        if (this.state.status !== BridgedClientStatus.CONNECTED) {
+            throw Error('Client is not connected');
+        }
+        return this.state.client as Client;
```

Comment:
Surprised you needed the `as`.

Reply:
I don't think it can now what `client` will be, since it depends on the `.status`, and we don't have a way (yet?) to represent that in the type system itself.